webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -462,52 +468,33 @@ export function getChannelTimezones(channelId: string): ActionFunc {
     };
 }
 
-export function fetchMyChannelsAndMembersREST(teamId: string): ActionFunc {
+export function fetchMyChannelsAndMembersREST(teamId: string): ActionFunc<{channels: ServerChannel[]; channelMembers: ChannelMembership[]}> {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        dispatch({
-            type: ChannelTypes.CHANNELS_REQUEST,
-            data: null,
-        });
-
         let channels;
         let channelMembers;
-        const state = getState();
-        const shouldFetchArchived = isMinimumServerVersion(getServerVersion(state), 5, 21);
         try {
             [channels, channelMembers] = await Promise.all([
-                Client4.getMyChannels(teamId, shouldFetchArchived),
+                Client4.getMyChannels(teamId),
                 Client4.getMyChannelMembers(teamId),
             ]);
         } catch (error) {
             forceLogoutIfNecessary(error, dispatch, getState);
-            dispatch({type: ChannelTypes.CHANNELS_FAILURE, error});
             dispatch(logError(error));
-            return {error};
+            return {error: error as ServerError};
         }
 
-        const {currentUserId} = state.entities.users;
-        const {currentChannelId} = state.entities.channels;
-
         dispatch(batchActions([
             {
                 type: ChannelTypes.RECEIVED_CHANNELS,
                 teamId,
                 data: channels,
-                currentChannelId,
-            },
-            {
-                type: ChannelTypes.CHANNELS_SUCCESS,
```

Comment:
I don't have enough context to understand why we are removing this. Can you expand on this?

Reply:
These network requests trackers monitor their loading states, such as SUCCESS, REQUEST, and FAILURE. Initially, these states were assigned to every request. However, it was observed that the cost of adding these states was not justified with the very few places we ended up using these. Therefore, a decision was made to remove these states eventually, as they were not providing significant benefits.

Reply:
Can you expand more on the nature of the "cost" of adding these states? More memory? More re-renders?

Reply:
It's more changes to the redux state which triggers it to update and do all the work to figure out if anything in the whole app needs to potentially re-render, so it's closer to a render. It's not going to impact practical performance, but it can make any problem code worse since it runs more often. Since this is a batched action, it shouldn't have any practical effect though.

We actually had a campaign to remove them a while back and this is just one of the ones that's lingered around. We've used that request state so rarely (mostly in early versions of the previous mobile app) that this is mostly a cleanup task

Reply:
@hmhealey I have not been able to find that epic on Jira. Can you please help me with that?